### PR TITLE
Archive in multiple jobs

### PR DESCRIPTION
**Pull request by mrcaseb**
Trying to set up muiltiple jobs by adding the release tags to archive into an array in the matrix setup

Needs test runs to see if it works. 

Main problem here: new releases need to be added manually to the array, otherwise they won't be archived

### CODE REVIEW

File: .github/workflows/run_archive.yaml [hunk 1]
```diff
@@ -19,13 +19,33 @@ jobs:
       matrix:
         config:
           - {os: ubuntu-20.04,   r: 'release'}
-
+        tag:
```
Comment (tanho63) [inline .github/workflows/run_archive.yaml:22]: we should change the `name` parameter on L15 to ${{ matrix.tag }}

Comment (mrcaseb) [reply .github/workflows/run_archive.yaml:22]: Ah yep, makes sense

**Comment by tanho63** [comment]
FYI: can test by changing which branch the workflow is coming from here:
 
image

File: .github/workflows/run_archive.yaml [hunk 2]
```diff
@@ -3,29 +3,50 @@ on:
...
+          - players
+          - players_components
+          - rosters
+          - snap_counts
+          # - test
+          - weekly_rosters
     env:
       GITHUB_PAT: ${{ secrets.NFLVERSE_GH_TOKEN }}
       R_KEEP_PKG_SOURCE: yes
+      NFLVERSE_ARCHIVE_TAG: ${{ matrix.tag }}
```
Comment (tanho63) [inline .github/workflows/run_archive.yaml:46]: can set env from actions rather than R

**Comment by mrcaseb** [comment]
We run into a pretty much useless warning 

> Warning message:
Failed to create release: "archive-2024-02-14" already exists!

I think that's here https://github.com/nflverse/nflverse-data/blob/0a74f43fb19785e1a85ae928a51aa887e7a4e34a/R/archive.R#L11-L16

**Comment by tanho63** [comment]
Okay I think we're good. I also noticed the release creation thing and I was worried we would run into race conditions where each step would try to create the release at the same time and fail, so I moved the release creation step into a prerequisite step before the archiving and made it `need` the release creation. Then I made the archive function check for the matching release name

**Comment by tanho63** [comment]
If we wanted to be maximally efficient we could probably use gh + bash instead of R + piggyback to download stuff but I don't wanna figure that out atm

**Comment by mrcaseb** [comment]
> Okay I think we're good. I also noticed the release creation thing and I was worried we would run into race conditions where each step would try to create the release at the same time and fail, so I moved the release creation step into a prerequisite step before the archiving and made it `need` the release creation. Then I made the archive function check for the matching release name

Looks good to me and seems solid. The only thing that worries me is that we have to manually maintain the list of releases to archive

**Comment by tanho63** [comment]
Agreed. Maybe we could figure out some kind of automated R script that builds the workflow file and commits it :joy: gotta jump in the car now

**Comment by mrcaseb** [comment]
Dropping links for dynamic matrix creation 

- https://www.kenmuse.com/blog/dynamic-build-matrices-in-github-actions/

**Comment by tanho63** [comment]
> https://www.kenmuse.com/blog/dynamic-build-matrices-in-github-actions/

oh shit this is cool, i can do gh release list and then output that to json and pass that into matrix

**Comment by mrcaseb** [comment]
> > https://www.kenmuse.com/blog/dynamic-build-matrices-in-github-actions/
> 
> oh shit this is cool, i can do gh release list and then output that to json and pass that into matrix

That's how I understood it but I don't know where the JSON must live

**Comment by tanho63** [comment]
action steps can have outputs that are referred to from other steps

**Pull request merged**